### PR TITLE
Add ability to filter sync queries

### PR DESCRIPTION
**Pull request by ssd71**
This PR adds a filter argument to the sync query in the datasync package

### CODE REVIEW

**Comment by wtrocki** [comment]
Checking

File: packages/graphback-datasync/src/DataSyncPlugin.ts [hunk 1]
```diff
@@ -60,7 +60,8 @@ export class DataSyncPlugin extends GraphbackPlugin {
                     [deltaQuery]: `${getDeltaListType(modelName)}!`
                 });
                 schemaComposer.Query.addFieldArgs(deltaQuery, {
-                    lastSync: 'String!'
+                    lastSync: 'String!',
+                    filter: `${getInputTypeName(modelName, GraphbackOperationType.FIND)}`
```
Comment (wtrocki) [inline packages/graphback-datasync/src/DataSyncPlugin.ts:64]: I will extract this to var

File: packages/graphback-datasync/package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
   },
   "dependencies": {
     "@graphback/core": "0.14.0-alpha3",
+    "@graphback/codegen-schema": "0.14.0-alpha3",
     "@graphback/runtime": "0.14.0-alpha3",
     "@graphback/runtime-knex": "0.14.0-alpha3",
```
Comment (wtrocki) [inline packages/graphback-datasync/package.json:43]: Why knex and mongo here. I will see only mongo

File: packages/graphback-datasync/tests/DataSyncMongoDBTests.ts [hunk 3]
```diff
@@ -108,4 +108,39 @@ describe('Soft deletion test', () => {
...
+    }
+
+    // Sync query
+    const deltaPosts = await Post.sync("0", {
+      text: {
+        startsWith: 'post'
+      }
+    });
+
+    expect(deltaPosts.length).toBeGreaterThanOrEqual(1);
```
Comment (machi1990) [inline packages/graphback-datasync/tests/DataSyncMongoDBTests.ts:139]: Should we change this assertion to:

```suggestion
    expect(deltaPosts.length).toEqual(2));
```

File: packages/graphback-datasync/tests/DataSyncMongoDBTests.ts [hunk 4]
```diff
@@ -108,4 +108,39 @@ describe('Soft deletion test', () => {
...
+
+    const { Post }  = context.providers;
+
+    // Create some posts
+    for (const postTitle of ["post", "post2", "trains"]) {
+      await Post.create({ text: postTitle });
+    }
+
+    // Sync query
+    const deltaPosts = await Post.sync("0", {
```
Comment (wtrocki) [inline packages/graphback-datasync/tests/DataSyncMongoDBTests.ts:133]: I would recommend to do second sync with return timestamp and check if result is zero etc. (maybe separate tests)

File: packages/graphback-datasync/tests/DataSyncPluginTest.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { readFileSync } from 'fs';
 import { buildSchema, printSchema } from 'graphql';
 import { GraphbackCoreMetadata } from '@graphback/core';
+import { SchemaCRUDPlugin } from "@graphback/codegen-schema";
```
Comment (craicoverflow) [inline packages/graphback-datasync/tests/DataSyncPluginTest.ts:4]: If this is just used in tests (looks that way) this should be a devDependency

File: packages/graphback-datasync/package.json [hunk 6]
```diff
@@ -38,8 +38,8 @@
   },
   "dependencies": {
     "@graphback/core": "0.14.0-alpha3",
+    "@graphback/codegen-schema": "0.14.0-alpha3",
```
Comment (craicoverflow) [inline packages/graphback-datasync/package.json:41]: If this is just used in tests (looks that way) this should be a devDependency

File: packages/graphback-datasync/tests/DataSyncPluginTest.ts (hunk 5 above)
Comment (ssd71) [reply packages/graphback-datasync/tests/DataSyncPluginTest.ts:4]: I think we actually need the schema plugin to do `@versioned` when using datasync

Comment (wtrocki) [reply packages/graphback-datasync/tests/DataSyncPluginTest.ts:4]: Yep. We can think of aggregating all 3 annotations into one as consolidation. WDYT? I can create an optional issue to deal with this.

**Comment by wtrocki** [comment]
Looking for @ssd71 to address some comments

File: packages/graphback-datasync/tests/DataSyncPluginTest.ts (hunk 5 above)
Comment (ssd71) [reply packages/graphback-datasync/tests/DataSyncPluginTest.ts:4]: I am onboard :+1:  with this!

File: packages/graphback-datasync/tests/DataSyncMongoDBTests.ts (hunk 4 above)
Comment (ssd71) [reply packages/graphback-datasync/tests/DataSyncMongoDBTests.ts:133]: On it

File: packages/graphback-datasync/tests/DataSyncPluginTest.ts (hunk 5 above)
Comment (craicoverflow) [reply packages/graphback-datasync/tests/DataSyncPluginTest.ts:4]: > I think we actually need the schema plugin to do @versioned when using datasync

Yes, but this happens independently of the datasync package in the plugin chain. The datasync package only uses it in test files, so it should be a devDependency

image

**Comment by wtrocki** [comment]
Ah really sad that did not make it to alpha

**Pull request merged**